mesonbuild/modules/gnome.py

```diff
@@ -789,10 +789,13 @@ def get_devenv(self) -> T.Optional[build.EnvironmentVariables]:
 
     def _get_gir_dep(self, state: 'ModuleState') -> T.Tuple[Dependency, T.Union[build.Executable, 'ExternalProgram', 'OverrideProgram'],
                                                             T.Union[build.Executable, 'ExternalProgram', 'OverrideProgram']]:
-        if not self.gir_dep:
-            self.gir_dep = state.dependency('gobject-introspection-1.0')
-            self.giscanner = state.find_tool('g-ir-scanner', 'gobject-introspection-1.0', 'g_ir_scanner')
-            self.gicompiler = state.find_tool('g-ir-compiler', 'gobject-introspection-1.0', 'g_ir_compiler')
+        if not self.gir_dep or not self.gir_dep.found():
+            for (dep, req) in [('gobject-introspection-no-typelibs-1.0', False), ('gobject-introspection-1.0', True)]:
```

Comment:
I think it is a bit more complicated because after g-i is done we want to switch to `gobject-introspection-1.0`. That way subprojects coming after g-i will have typelibs dependencies.

That means you should invert the order and check for `gobject-introspection-1.0` first. If it is -no-typelib that has been found you should remember that and re-check for `gobject-introspection-1.0` next time.

Reply:
@ylatuya I think the new version of your patch still have the same issue because it re-check the dependency only `if not self.gir_dep:`

Reply:
Yes, I pushed it before reading your comments.

Reply:
It should be handled correctly now